yt_dlp/extractor/telegram.py

```diff
@@ -1,41 +1,141 @@
-        self._sort_formats(formats)
-
-        return {
-            'id': video_id,
-            'title': self._html_search_meta(['og:title', 'twitter:title'], webpage, default=None),
-            'description': self._html_search_meta(
-                ['og:description', 'twitter:description'], webpage,
-                default=clean_html(get_element_by_class('tgme_widget_message_text', webpage_embed))),
-            'thumbnail': self._search_regex(
-                r'tgme_widget_message_video_thumb"[^>]+background-image:url\(\'([^\']+)\'\)',
-                webpage_embed, 'thumbnail'),
-            'formats': formats,
+        query = {'embed': '1'}
+        if self.get_param('noplaylist'):
+            query['single'] = True
+        elif self.get_param('noplaylist') is False:
+            pass
+        elif '?single' in url:  # param 'noplaylist' is None
+            query['single'] = True
+        url = update_url_query(url, {'single': []})  # strip 'single' from original URL query
+        embed = self._download_webpage(url, msg_id, query=query, note='Downloading embed frame')
+
+        message = {
+            'title': clean_html(get_element_by_class('tgme_widget_message_author', embed)),
+            'description': clean_html(get_element_by_class('tgme_widget_message_text', embed)),
+            'uploader': uploader,
+            'timestamp': unified_timestamp(self._search_regex(
+                r'<time[^>]*datetime="([^"]*)"', embed, 'timestamp', fatal=False)),
         }
+
+        videos = []
+        for video in re.findall(r'<a class="tgme_widget_message_video_player(?s:.+?)</time>', embed):
+            video_url = self._search_regex(
+                r'<video[^>]+src="([^"]+)"', video, 'video URL', fatal=False)
+            webpage_url = self._search_regex(
+                r'<a class="tgme_widget_message_video_player[^>]+href="([^"]+)"',
+                video, 'webpage URL', fatal=False)
+            if not video_url or not webpage_url:
+                continue
+            formats = [{
+                'url': video_url,
+                'ext': 'mp4',
+            }]
+            self._sort_formats(formats)
+            videos.append({
+                'id': url_basename(webpage_url),
+                'duration': parse_duration(self._search_regex(
+                    r'<time[^>]+duration[^>]*>([\d:]+)</time>', video, 'duration', fatal=False)),
+                'thumbnail': self._search_regex(
+                    r'tgme_widget_message_video_thumb"[^>]+background-image:url\(\'([^\']+)\'\)',
+                    video, 'thumbnail', fatal=False),
+                'formats': formats,
+                **message,
+            })
+
+        if query.get('single') or len(videos) < 2:
+            return traverse_obj(videos, lambda _, x: x['id'] == msg_id, get_all=False)
+        else:
+            return self.playlist_result(
+                videos, f'{uploader}-{msg_id}', f'{message["title"]} {msg_id}', message['description'])
```

Comment:
title/description should be non-fatal

Reply:
`get_element_by_class` is non-fatal and the `title`/`description` keys of the `message` dict will always exist. or am I missing something?

Reply:
u r right, my bad

Reply:
```suggestion
                videos, f'{uploader}-{msg_id}',
                format_field(message, title, f'%s {msg_id}'), message['description'])
```
 to set title=None if message[title] is None